fvm/fvm_test.go

```diff
@@ -3504,6 +3504,43 @@ func TestSettingExecutionWeights(t *testing.T) {
 			assert.True(t, errors.IsComputationLimitExceededError(tx.Err))
 		},
 	))
+
+	t.Run("transaction allocating high memory should fail", newVMTest().withBootstrapProcedureOptions(
+		fvm.WithMinimumStorageReservation(fvm.DefaultMinimumStorageReservation),
+		fvm.WithAccountCreationFee(fvm.DefaultAccountCreationFee),
+		fvm.WithStorageMBPerFLOW(fvm.DefaultStorageMBPerFLOW),
+		fvm.WithExecutionMemoryWeights(weightedMeter.DefaultMemoryWeights),
+	).withContextOptions(
+		fvm.WithMemoryLimit(10_000_000_000),
```

Comment:
Would it make sense to have a much lower value as the memory limit, to avoid this test case from consuming lot of memory?